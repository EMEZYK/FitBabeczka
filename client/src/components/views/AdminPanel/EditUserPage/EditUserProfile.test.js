import { render, cleanup } from "@testing-library/react";
import { screen } from "@testing-library/dom";
import "@testing-library/jest-dom";
import EditProfilePage from "./EditUserProfilePage";

describe("AboutMePage components", () => {
  afterEach(cleanup);

  test("Check if takes a snapshot',", () => {
    const { asFragment } = render(<EditProfilePage />);
    expect(asFragment(<EditProfilePage />)).toMatchSnapshot();
  });

  test("Check if renders form title", () => {
    render(<EditProfilePage />);
    const title = screen.getByText("Zmień hasło");
    expect(title).toBeInTheDocument();
  });

  test("Check if password and confirmPassword inputs are required", () => {
    render(<EditProfilePage />);
    const passwordInput = screen.queryByTestId("password");
    expect(passwordInput).toBeRequired();

    const confirmPasswordInput = screen.queryByTestId("confirm-password");
    expect(confirmPasswordInput).toBeRequired();
  });
});
