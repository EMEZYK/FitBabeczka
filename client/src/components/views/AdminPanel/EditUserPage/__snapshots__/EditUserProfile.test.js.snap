// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutMePage components Check if takes a snapshot', 1`] = `
<DocumentFragment>
  <div
    class="sc-bczRLJ hvLzQT"
  >
    <div
      class="sc-bczRLJ sc-dkzDqf lhmkud kDgsQz"
    >
      <h2
        class="sc-hKMtZM gGDAVZ"
      >
        Zmień hasło
      </h2>
      <form
        class="sc-eCYdqJ brYmts"
        type="submit"
      >
        <label
          class="sc-jSMfEi eHcCgu"
          for="email"
        >
          Email
        </label>
        <input
          class="sc-ftvSup EXDDb"
          id="email"
          required=""
          type="email"
          value=""
        />
        <label
          class="sc-jSMfEi eHcCgu"
          for="password"
        >
          Hasło
        </label>
        <input
          class="sc-ftvSup EXDDb"
          data-testid="password"
          id="password"
          required=""
          type="password"
          value=""
        />
        <label
          class="sc-jSMfEi eHcCgu"
          for="password"
        >
          Potwierdź hasło
        </label>
        <input
          class="sc-ftvSup EXDDb"
          data-testid="confirm-password"
          id="confirmPassword"
          required=""
          type="password"
          value=""
        />
        <button
          class="sc-gsnTZi sc-gKXOVf goHXkf dEJMjr"
          type="submit"
        >
          Zapisz
        </button>
        <div
          class="Toastify"
        />
      </form>
    </div>
  </div>
</DocumentFragment>
`;
